api-wrapper/services/api-wrapper/src/main/java/net/catenax/edc/apiwrapper/security/BasicAuthenticationService.java

```diff
@@ -44,14 +49,15 @@ private boolean checkBasicAuthValid(Result<BasicAuthCredentials> basicAuthCreden
         var credentials = basicAuthCredentialsResult.getContent();
         var username = credentials.username;
         var password = credentials.password;
-        var password4Username = users.get(username);
 
-        if (password4Username == null || !password4Username.equals(password)) {
-            monitor.debug("Basic auth user could not be found or password wrong");
-            return false;
-        }
+        Predicate<BasicAuthVaultLabels> isCorrectUser = e -> e.getUsername().equals(username);
+        Predicate<BasicAuthVaultLabels> isCorrectVaultKey = e -> vault.resolveSecret(e.getVaultKey()).equals(password);
 
-        return true;
+        return this.basicAuthVaultLabels.stream()
+                .filter(isCorrectUser)
+                .filter(isCorrectVaultKey)
+                .findAny()
+                .isPresent();
```

Comment:
IntelliJ says, that this is shorter 😅

```suggestion
                .anyMatch(isCorrectVaultKey);
```